### PR TITLE
use short config syntax for factories

### PR DESCRIPTION
**Pull request by snoek09**
This fixes https://github.com/symfony/symfony-docs/issues/7208.

### CODE REVIEW

File: service_container/factories.rst [hunk 1]
```diff
@@ -106,6 +106,23 @@ configure the service container to use the
     the configured class name may be used by compiler passes and therefore
     should be set to a sensible value.
 
+.. note::
+
+    The traditional configurator syntax in YAML files used an array to define
```
Comment (xabbuh) [inline service_container/factories.rst]: configuration syntax

File: service_container/factories.rst [hunk 2]
```diff
@@ -106,6 +106,23 @@ configure the service container to use the
...
+
+        app.newsletter_manager:
+            # new syntax
+            factory: 'app.newsletter_manager_factory:createNewsletterManager'
+            # old syntax
+            factory: ['@app.newsletter_manager_factory', createNewsletterManager]
+
+    .. versionadded:: 3.2
+        The ``factory_service:method_name`` syntax for the YAML configuration format
+        was introduced in Symfony 3.2.
```
Comment (snoek09) [inline service_container/factories.rst]: I'll change the PR to merge  to 2.8 instead.

**Comment by wouterj** [comment]
Thanks @snoek09!

👍
status: reviewed

File: service_container/factories.rst [hunk 3]
```diff
@@ -106,6 +106,23 @@ configure the service container to use the
...
+
+        app.newsletter_manager:
+            # new syntax
+            factory: 'app.newsletter_manager_factory:createNewsletterManager'
+            # old syntax
+            factory: ['@app.newsletter_manager_factory', createNewsletterManager]
+
+    .. versionadded:: 2.6
+        The ``factory_service:method_name`` syntax for the YAML configuration format
+        was introduced in Symfony 2.6.
```
Comment (xabbuh) [inline service_container/factories.rst]: The `versionadded` directive is not needed here.

**Review by javiereguiluz — APPROVED** [review]
👍

**Comment by xabbuh** [comment]
:+1:

**Comment by xabbuh** [comment]
Thank you @snoek09.

**Pull request closed without merging**